### PR TITLE
DarkHex & Hex Fixes and Updates

### PR DESCRIPTION
**Pull request by BedirT**
- Error on player board representation fixed (no flood fill applied; missing info)
- Fixed naming convention for the parameters, common term is "number of rows" rather than "column size" etc.
- Reduced the default board size to a more reasonable value.
- Fixed game logic for hex
- Board size with a row or col 2 fixed

### CODE REVIEW

**Comment by BedirT** [comment]
I think I am getting a precision error on playthroughs on the push

**Comment by lanctot** [comment]
> I think I am getting a precision error on playthroughs on the push

Can you try reverting python_mfg_predator_prey.txt ?

**Comment by lanctot** [comment]
Hi Bedir,

Can you elaborate on the bug?

Is it easy to send us a serialized state string or action history where some demonstrably incorrect happens?

**Comment by BedirT** [comment]
Yeah sure thing. I realized the bug on the test I added to ```hex_test.cc```.

In Hex and DarkHex we do want to evaluate uneven boards at times, and that is why I converted the board sizes to num_cols and num_rows before, but I actually did it in a way that it will always be larger on the num_cols then num_rows (board looking cleaner) which was problematic since we were printing the other way and checking for terminal state using the unaltered position rather than altered version.

Here is the output where it clearly shows that the connection is not properly satisfied (due to row-column sizing issue I mentioned):

```
. y p 
 . y p 
  . z p 
   . z . 
Spiel Fatal Error: /home/bedir/Documents/open_spiel/open_spiel/games/hex_test.cc:39 CHECK_TRUE(state->IsTerminal())
```

And here is the output where it is now fixed.

```
. y q 
 . y q 
  . y q 
   . X . 
```

This actually causes the game to never reach a terminal state at times (which causes infinite game time...) I am not sure how did this slip my tests before :)

**Comment by BedirT** [comment]
@lanctot any update on this?

**Comment by lanctot** [comment]
Yeah, sorry, I do intend to go through it with @TWAnthony tomorrow and hopefully pull it in this week.

**Comment by BedirT** [comment]
No worries that's perfect :)

File: open_spiel/games/dark_hex.cc [hunk 1]
```diff
@@ -131,7 +130,21 @@ void DarkHexState::DoApplyAction(Action move) {
...
+  } else if (state_.BoardAt(move) == CellState::kWhite ||
+             state_.BoardAt(move) == CellState::kWhiteEast ||
+             state_.BoardAt(move) == CellState::kWhiteWest) {
+    cur_view[move] = CellState::kWhite;
+  } else if (state_.BoardAt(move) == CellState::kBlackWin ||
+             state_.BoardAt(move) == CellState::kWhiteWin) {
+    cur_view[move] = state_.BoardAt(move);
+  } else {
+    SPIEL_CHECK_TRUE(false);
+  }
```
[the remark this answers is not in the document]
Comment (lanctot) [reply open_spiel/games/dark_hex.cc:147]: And also: why is that not already happening in the InformationStateString (or ObservationString) functions?

[the remark this answers is not in the document]
Comment (lanctot) [reply open_spiel/games/dark_hex.cc:147]: Ah right, ok I think it is fine.

**Pull request merged**